gprofiler/application_separators.py

```diff
@@ -0,0 +1,193 @@
+
+
+class UwsgiApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "uwsgi"
+
+    def is_supported(self, process: Process) -> bool:
+        return process.cmdline()[0] == "uwsgi"
+
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        config_file = None
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        wsgi = get_positional_arg(process.cmdline(), "-w")
+        if wsgi is not None:
+            return wsgi
+
+        wsgi = self._find_wsgi_from_config_file(process)
+        if wsgi is not None:
+            return wsgi
+
+        logger.warning("Couldn't find uwsgi wsgi module, both from cmdline and from config file")
+        return None
+
+
+class CeleryApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "celery"
+
+    def is_supported(self, process: Process) -> bool:
+        if process.cmdline()[0] == "celery":
+            return True
+
+        return len(process.cmdline()) >= 3 and ["-m", "celery"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        app_name = get_positional_arg(process.cmdline(), "-A") or get_positional_arg(process.cmdline(), "--app")
+        if app_name is None:
+            logger.warning("Couldn't find positional argument -A or --app for application indication")
+            return None
+
+        return app_name
+
+
+class PySparkApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "pyspark"
+
+    def is_supported(self, process: Process) -> bool:
+        # We're looking for pythonXX -m pyspark.daemon
+        return process.cmdline() >= 3 and ["-m", "pyspark.daemon"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        return "generic pyspark.daemon"
+
+
+class PythonModuleApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "generic-python"
+
+    def is_supported(self, process: Process) -> bool:
+        if "python" not in process.cmdline()[0]:
+            return False
+
+        return (len(process.cmdline()) > 3 and process.cmdline()[1] == "-m") or (
+            len(process.cmdline()) >= 2 and process.cmdline()[1].endswith(".py")
```

Comment:
The ".py" file can be passed anywhere in the command line. I think you can just check if `any(arg.endswith(".py") for arg in process.cmdline()[1:])`

Reply:
You're right but  it can also be passed in non python application (some wrapper scripts and so..), here I'm checking for running python script in the most simple way as I don't want to get into more complicated scenarios and maybe produce false positives, as this case will cover most of the scenarios (the most common scenarios are running a python script with `-m` or with just python script as argv[1])

Reply:
It's okay to ignore other cases now. However if you can extract this logic (from here, PySpark and possibly others) to some function that will currently check only for `argv[0] == python && argv[1] == -m` for example, and later on we can make it smarter and skip other flags etc.

Reply:
My suggestion was done, @YishaiZinkin 's point will not be addressed as part of the MVP; resolving.